File: Example/Classes/Control/TGInitialViewController.m
//
//  TGInitialViewController.m
//  TGCameraViewController
//
//  Created by Bruno Furtado on 15/09/14.
//  Copyright (c) 2014 Tudo Gostoso Internet. All rights reserved.
//

#import "TGInitialViewController.h"
#import "TGAlbum.h"
#import "TGCameraNavigationController.h"

@interface TGInitialViewController ()

@property (strong, nonatomic) IBOutlet UIImageView *photoView;

- (IBAction)takePhotoTapped;
- (IBAction)chooseExistingPhotoTapped;

- (void)clearTapped;

@end



@implementation TGInitialViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    _photoView.clipsToBounds = YES;
    
    UIBarButtonItem *clearButton = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCancel
                                                                                 target:self
                                                                                 action:@selector(clearTapped)];
    
    self.navigationItem.rightBarButtonItem = clearButton;
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
}

#pragma mark -
#pragma mark - TGCameraDelegate

- (void)cameraDidTakePhoto:(UIImage *)image
{
    _photoView.image = image;
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (void)cameraDidCancel
{
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (void)cameraWillTakePhoto
{
    NSLog(@"%s", __PRETTY_FUNCTION__);
}

#pragma mark -
#pragma mark - UIImagePickerControllerDelegate

- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary *)info
{
    _photoView.image = [TGAlbum imageWithMediaInfo:info];
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (void)imagePickerControllerDidCancel:(UIImagePickerController *)picker
{
    [self dismissViewControllerAnimated:YES completion:nil];
}

#pragma mark -
#pragma mark - Actions

- (IBAction)takePhotoTapped
{    
    TGCameraNavigationController *navigationController = [TGCameraNavigationController newWithCameraDelegate:self];
    [self presentViewController:navigationController animated:YES completion:nil];
}

- (IBAction)chooseExistingPhotoTapped
{
    UIImagePickerController *pickerController = [TGAlbum imagePickerControllerWithDelegate:self];
    [self presentViewController:pickerController animated:YES completion:nil];
}

#pragma mark -
#pragma mark - Private methods

- (void)clearTapped
{
    _photoView.image = nil;
}

@end

File: ExampleSwift/TGInitialViewController.swift
//
//  ViewController.swift
//  ExampleSwift
//
//  Created by Mario Cecchi on 7/20/16.
//  Copyright © 2016 Tudo Gostoso Internet. All rights reserved.
//

import UIKit

class TGInitialViewController: UIViewController, TGCameraDelegate {
    @IBOutlet weak var photoView: UIImageView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // set custom tint color
        //TGCameraColor.setTintColor(.greenColor())
        
        // save image to album
        TGCamera.setOption(kTGCameraOptionSaveImageToAlbum, value: true)
        
        // hide switch camera button
        //TGCamera.setOption(kTGCameraOptionHiddenToggleButton, value: true)
        
        // hide album button
        //TGCamera.setOption(kTGCameraOptionHiddenAlbumButton, value: true)
        
        // hide filter button
        //TGCamera.setOption(kTGCameraOptionHiddenFilterButton, value: true)
        
        photoView.clipsToBounds = true
        
        let clearButton = UIBarButtonItem(barButtonSystemItem: .Cancel, target: self, action:#selector(clearTapped))
        navigationItem.rightBarButtonItem = clearButton
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    
    // MARK: TGCameraDelegate - Required methods
    
    func cameraDidCancel() {
        dismissViewControllerAnimated(true, completion: nil)
    }
    
    func cameraDidTakePhoto(image: UIImage!) {
        photoView.image = image
        dismissViewControllerAnimated(true, completion: nil)
    }
    
    func cameraDidSelectAlbumPhoto(image: UIImage!) {
        photoView.image = image
        dismissViewControllerAnimated(true, completion: nil)
    }
    
    
    // MARK: TGCameraDelegate - Optional methods
    
    func cameraWillTakePhoto() {
        print("cameraWillTakePhoto")
    }
    
    func cameraDidSavePhotoAtPath(assetURL: NSURL!) {
        print("cameraDidSavePhotoAtPath: \(assetURL)")
    }
    
    func cameraDidSavePhotoWithError(error: NSError!) {
        print("cameraDidSavePhotoWithError \(error)")
    }
    
    
    // MARK: Actions
    
    @IBAction func takePhotoTapped() {
        let navigationController = TGCameraNavigationController.newWithCameraDelegate(self)
        presentViewController(navigationController, animated: true, completion: nil)
    }
    
    
    // MARK: Private methods
    
    func clearTapped() {
        photoView.image = nil
    }
}

